nats.go

```diff
@@ -2692,7 +2695,10 @@ func (nc *Conn) processMsg(data []byte) {
 	}
 
 	// Copy them into string
-	subj := string(nc.ps.ma.subject)
+	subj := sub.Subject
+	if sub.subjectMayDiffer {
```

Comment:
But again, not enough for the above `subj := sub.Subject`.. I am sorry to be a pain, but I am just stating that the way it is right now, you may still get DATA RACE report one day because of that.
So maybe something like:
```
var subj string
if sub.subjectMayDiffer {
    subj = string(nc.pas.ma.subject)
} else {
   subj = sub.Subject
}
```

Reply:
hmm, it's just a way to get rid of race condition checker message, because of I don't see any usage or changing Subject of subscription?

Reply:
Yes we do, look for resetOrderedConsumer()...

Reply:
Yes, but previous string already allocated and attached to subj anyway

Reply:
We are ok because indeed the resetOrderSubject() is invoked only from this same go routine and after that line. Ok, we are good.

Reply:
```go
var g Subscription
g.Subject = "fun"
k := g.Subject
g.Subject = "not fun"
println(k) // fun
```

Reply:
Oh, this is not what I was pointing out, what I was afraid is that resetOrderConsumer() could be invoked from a different go routine, which apparently and as of now is not. But if it was, you would basically have a situation where processMsg() from thread 1 would be doing `subj := sub.Subject` while thread 2 would be doing `sub.Subject = newSubject`. But again, seems like resetOrderConsumer() is invoked from the processMsg() go routine only, and after you capture the subject.